projects/angular/src/button/_buttons.clarity.scss

```diff
@@ -61,6 +61,14 @@
     @return lookupFromDangerOutlineButtonColors($property);
   }
 
+  @if $type == warning {
```

Comment:
This type was missing, but was being called from later in this file which is what was causing the warning styles to match the primary color in dark theme. 

https://github.com/vmware-clarity/ng-clarity/blob/7e501913a580d4ea5cf84bd8b68167c322316090/projects/angular/src/button/_buttons.clarity.scss#L321